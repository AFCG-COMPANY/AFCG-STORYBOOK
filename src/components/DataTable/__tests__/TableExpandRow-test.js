import React from 'react';
import { mount } from 'enzyme';
import { Table, TableBody, TableExpandRow, TableExpandedRow } from '../';

describe('DataTable.TableExpandRow', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      className: 'custom-class',
      isExpanded: false,
      onExpand: jest.fn(),
      ariaLabel: 'Aria label',
    };
  });

  it('should render', () => {
    const wrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} />
          <TableExpandedRow />
        </TableBody>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should initially not define `data-previous-value`', () => {
    const wrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} />
          <TableExpandedRow />
        </TableBody>
      </Table>
    );
    expect(
      Object.keys(wrapper.find('TableCell').props()).indexOf(
        'data-previous-value'
      ) !== -1
    ).toBe(true);
    expect(
      wrapper.find('TableCell').prop('data-previous-value')
    ).not.toBeDefined();
  });

  it('should expand when the expand button is clicked', () => {
    const initialWrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} />
          <TableExpandedRow />
        </TableBody>
      </Table>
    );
    initialWrapper.find('button').simulate('click');
    expect(mockProps.onExpand).toHaveBeenCalledTimes(1);

    const expandedWrapper = mount(
      <Table>
        <TableBody>
          <TableExpandRow {...mockProps} isExpanded />
          <TableExpandedRow />
        </TableBody>
      </Table>
    );
    expect(expandedWrapper.find('TableCell').prop('data-previous-value')).toBe(
      'collapsed'
    );
  });
});
