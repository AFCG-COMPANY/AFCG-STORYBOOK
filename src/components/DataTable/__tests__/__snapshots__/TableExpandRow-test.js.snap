// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableExpandRow should render 1`] = `
<Table>
  <table
    className="bx--data-table-v2 bx--data-table-v2--zebra"
  >
    <TableBody>
      <tbody>
        <TableExpandRow
          ariaLabel="Aria label"
          className="custom-class"
          isExpanded={false}
          onExpand={[MockFunction]}
        >
          <tr
            className="bx--parent-row-v2 custom-class"
            data-parent-row={true}
          >
            <TableCell
              className="bx--table-expand-v2"
            >
              <td
                className="bx--table-expand-v2"
              >
                <button
                  aria-label="Aria label"
                  className="bx--table-expand-v2__button"
                  onClick={[MockFunction]}
                >
                  <Icon
                    className="bx--table-expand-v2__svg"
                    description="Provide a description that will be used as the title"
                    fillRule="evenodd"
                    name="chevron--right"
                    role="img"
                  >
                    <svg
                      alt="Provide a description that will be used as the title"
                      aria-label="Provide a description that will be used as the title"
                      className="bx--table-expand-v2__svg"
                      fillRule="evenodd"
                      height="12"
                      name="chevron--right"
                      role="img"
                      viewBox="0 0 7 12"
                      width="7"
                    >
                      <title>
                        Provide a description that will be used as the title
                      </title>
                      <path
                        d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
                        key="key0"
                      />
                    </svg>
                  </Icon>
                </button>
              </td>
            </TableCell>
          </tr>
        </TableExpandRow>
        <TableExpandedRow>
          <tr
            className="bx--expandable-row-v2"
            data-child-row={true}
          />
        </TableExpandedRow>
      </tbody>
    </TableBody>
  </table>
</Table>
`;
